scripts/common/actor.js

```diff
@@ -20,17 +20,38 @@ export class AgeOfSigmarActor extends Actor {
 
     prepareData() {
         super.prepareData();
+
         if (this.type === "player" || this.type === "npc") {
-            this._initializeData();
+
+            //defaults for Players
+            let flags = {
+                autoCalcToughness : true,
+                autoCalcMettle : true,
+                autoCalcWounds : true,
+                isSwarm : false
+            };
+
+            if(this.type === "npc") {
+                flags.autoCalcToughness = this.bio.type > 1;
+                flags.autoCalcMettle = this.bio.type > 2;
+                flags.autoCalcWounds = this.bio.type > 3;
+                flags.isSwarm = this.bio.type === 0;
+
+                if(game.settings.get("age-of-sigmar-soulbound", "sizeTokens")) {
+                    this._sizeToken(flags.isSwarm);
+                }                
+            }
```

Comment:
Is there a reason `flags` is this local object instead of something attached to actor data (like saved in the flags object)? You wouldn't need to pass it around everywhere if this were the case.

Reply:
It is only ever needed in this context for now. It might make sense if those flags where ever needed anywhere else but as it stance all they influence are the calculations that directly follow it